src/features/Instructors/InstructorsDetailPage/index.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useEffect, useRef } from 'react';
+import { Link, useParams, useHistory } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Container,
+  Pagination,
+  Tabs,
+  Tab,
+} from '@edx/paragon';
+import Table from 'features/Main/Table';
+import { fetchClassesData } from 'features/Classes/data/thunks';
+import { resetClassesTable, updateCurrentPage } from 'features/Classes/data/slice';
+import { fetchInstructorsData } from 'features/Instructors/data/thunks';
+import { columns } from 'features/Instructors/InstructorDetailTable/columns';
+import { initialPage } from 'features/constants';
+
+import 'features/Instructors/InstructorsDetailPage/index.scss';
+
+const InstructorsDetailPage = () => {
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const { instructorUsername } = useParams();
+
+  const institutionRef = useRef(undefined);
+  const [currentPage, setCurrentPage] = useState(initialPage);
+
+  const defaultInstructorInfo = {
+    instructorUsername: '',
+    instructorName: '',
+  };
+
+  const institution = useSelector((state) => state.main.selectedInstitution);
+  const classes = useSelector((state) => state.classes.table);
+  const instructorInfo = useSelector((state) => state.instructors.table.data)
+    .find((instructor) => instructor?.instructorUsername === instructorUsername) || defaultInstructorInfo;
+
+  const handlePagination = (targetPage) => {
+    setCurrentPage(targetPage);
+    dispatch(updateCurrentPage(targetPage));
+  };
+
+  useEffect(() => {
+    const initialState = {
+      results: [],
+      count: 0,
+      numPages: 0,
+    };
+
+    if (institution.id) {
+      dispatch(fetchClassesData(institution.id, initialPage, '', { instructor: instructorInfo.instructorUsername }));
+      dispatch(fetchInstructorsData(institution.id, initialPage, instructorInfo.instructorUsername));
+    }
+
+    return () => {
+      dispatch(resetClassesTable(initialState));
```

Comment:
you are not using the payload in the reducer definition so you can remove the initialState here

Reply:
@AuraAlba sorry, I forgot to remove it.